router/batchrouter/handle_observability.go

```diff
@@ -150,37 +193,34 @@ func (brt *Handle) emitAsyncEventDeliveryTimeMetrics(sourceID, destinationID str
 		}
 
 		// Get original job parameters for this job
-		originalParams, exists := asyncDestStruct.OriginalJobParameters[status.JobID]
```

Comment:
it is not clear what was wrong with the previous implementation and it is also not clear why what we are doing now is better

Reply:
I don't understand why this solution can really fix the issue. Even if it fixes some reporting issues, this early reset that you mentioned, doesn't it cause other issues too?
I guess this is a long-standing issue, so if it has been with us for so long, it shouldn't be that urgent. Why don't we spend the effort to fix it properly?

Reply:
this is fix only for emitting delivery time metrics for async destinations, it wasn't getting emitted as it is getting reset after upload.

Reply:
doing this as regular release.